devices/ePaper/Buffers/FrameBuffer1BitPerPixel.cs

```diff
@@ -0,0 +1,146 @@
+﻿// Copyright (c) 2022 The nanoFramework project contributors
+// See LICENSE file in the project root for full license information.
+
+using Iot.Device.ePaper.Primitives;
+
+namespace Iot.Device.ePaper.Buffers
+{
+    /// <summary>
+    /// A display frame buffer implementation for dual-color displays.
+    /// </summary>
+    public class FrameBuffer1BitPerPixel : FrameBufferBase
+    {
+        /// <inheritdoc/>
+        public override ColorFormat ColorFormat { get; } = ColorFormat.Color1BitPerPixel;
+
+        /// <summary>
+        /// Initializes a new instance of <see cref="FrameBuffer1BitPerPixel"/> class.
+        /// </summary>
+        /// <param name="height">The height of the frame to manage.</param>
+        /// <param name="width">The width of the frame to manage.</param>
+        public FrameBuffer1BitPerPixel(int height, int width)
+            : base(height, width)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of <see cref="FrameBuffer1BitPerPixel"/> class by copying the specified buffer.
+        /// </summary>
+        /// <param name="height">The height of the frame to manage.</param>
+        /// <param name="width">The width of the frame to manage.</param>
+        /// <param name="buffer">The starting frame buffer.</param>
+        public FrameBuffer1BitPerPixel(int height, int width, byte[] buffer)
+            : base(height, width, buffer)
+        {
+        }
+
+        /// <inheritdoc/>>
+        public override void Clear(Color color)
+        {
+            var colorValue = color.Color1bpp;
+            for (var i = 0; i < this.Buffer.Length; i++)
+                this.Buffer[i] = colorValue;
+        }
+
+        /// <inheritdoc/>
+        public override void Fill(Point start, int width, int height, Color color)
+        {
+            if (!this.IsPointWithinFrameBuffer(start))
+                return;
```

Comment:
we do use explicit brakets even for a single line. this does reduce problems.
I'm surprised it is not catched up by style cop!